### ISSUE TITLE
[SPIR-V] error: more than one unbounded resource

### ISSUE THREAD

**Issue by lxcug** [microsoft/DirectXShaderCompiler#7876]
I bind two unbound resources in same slot, the after is just an alias, it works with compiling spirv, but fails with compiling dxil

```
Buffer<float4>      BindlessFloatBuffers[]         : register(t0, space3);  
Buffer<uint4>       BindlessUIntBuffers[]          : register(t0, space3);
```

**Comment by s-perron** [comment]
In general, this is not disallowed in SPIR-V: https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#interfaces-resources-setandbinding.

We do not give an error before optimization because it is a common pattern to declare multiple resources but have all but one optimized away.

The SPIR-V validator does issue an error (as it does for DXIL), because the validation rule is tied to the type of the resource on the API side and can only be determined at runtime. I have not tested, but I believe the Vulkan validation layer would issue an error or warning if there is a problem.

**Comment by s-perron** [comment]
If your issue is that you want DXIL to work like spir-v does, reopen, and change the title to `[DXIL]`. I don't think they will do anything either. It would require changing DXIL's validation rules.

**Comment by scoluou** [comment]
> In general, this is not disallowed in SPIR-V: https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#interfaces-resources-setandbinding.
> 
> We do not give an error before optimization because it is a common pattern to declare multiple resources but have all but one optimized away.
> 
> The SPIR-V validator does issue an error (as it does for DXIL), because the validation rule is tied to the type of the resource on the API side and can only be determined at runtime. I have not tested, but I believe the Vulkan validation layer would issue an error or warning if there is a problem.

Actually, i create vulkan descriptor set layout and descriptor set with only one binding using the first "BindlessFloatBuffers" declartion, in hlsl the other declartion "BindlessUIntBuffers" is just an alias(i want to bind different types of buffers on the same slot), since hlsl doesn't support cast Buffer float4 to Buffer uint4.
